### PR TITLE
GH-797 Adds table colors to the sample themes

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes GH-797

@GordonSmith @dtsnell4 @mlzummo Please Review

https://rawgit.com/jbrundage/Visualization/TableColors_1.8.0/demos/dermatology.html?src/other/Table

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

File: src/other/Table.js [hunk 1]
```diff
@@ -73,12 +72,34 @@
...
+    Table.prototype.publish("tbodyRowBackgroundColor", null, "html-color", "Table body row color", null, { tags: ["Basic"] });
+    Table.prototype.publish("tbodyFirstColFontColor", null, "html-color", "Table body first column font color", null, { tags: ["Basic"] });
+    Table.prototype.publish("tbodyFirstColBackgroundColor", null, "html-color", "Table body first column background color", null, { tags: ["Basic"] });
+    
+    Table.prototype.publish("tbodyHoverRowFontColor", "#FFF", "html-color", "Table body hover row font color", null, { tags: ["Basic"] });
+    Table.prototype.publish("tbodyHoverRowBackgroundColor", null, "html-color", "Table body hover row background color", null, { tags: ["Basic"] });
+    
+    Table.prototype.publish("tbodySelectedRowFontColor", "#FFF", "html-color", "Table body selected row color", null, { tags: ["Basic"] });
+    Table.prototype.publish("tbodySelectedRowBackgroundColor", "#F48A00", "html-color", "Table body selected row color", null, { tags: ["Basic"] });
+
```
Comment (GordonSmith) [inline src/other/Table.js]: 1.  I am not sure these are all Basic?
2.  This approach will break existing CSS driven changes (which may be broke anyway) - if you look at https://github.com/hpcc-systems/Visualization/pull/832/files I have added "optional" support for publish params, with that you can default to null for each of these which will use the default CSS and then you can change them at the dermatology level (or the CSS level as appropriate.

**Comment by mlzummo** [comment]
red-bordered throws and error

**Comment by jbrundage** [comment]
@GordonSmith Added ext.optional to all of the newly added Table styling publish params. As for their "Basic" status... I've been considering the "Basic" classification to apply to any publish params that meet these two criteria:
1) this param can be easily understood by even the newest users 
2) this param would be commonly used when modifying this widget
I wasn't sure which (if any) of these newly added params should be considered something other than "Basic".

@mlzummo Fixed, thanks

**Comment by GordonSmith** [comment]
@jbrundage the row highlighting for selected rows is broken in "default".

File: src/other/Table.js [hunk 2]
```diff
@@ -73,12 +72,34 @@
...
+    Table.prototype.publish("tbodyFirstColBackgroundColor", null, "html-color", "Table body first column background color", null, { tags: ["Basic"], optional: true });
+    
+    Table.prototype.publish("tbodyHoverRowFontColor", null, "html-color", "Table body hover row font color", null, { tags: ["Basic"], optional: true });
+    Table.prototype.publish("tbodyHoverRowBackgroundColor", null, "html-color", "Table body hover row background color", null, { tags: ["Basic"], optional: true });
+    
+    Table.prototype.publish("tbodySelectedRowFontColor", null, "html-color", "Table body selected row color", null, { tags: ["Basic"], optional: true });
+    Table.prototype.publish("tbodySelectedRowBackgroundColor", null, "html-color", "Table body selected row color", null, { tags: ["Basic"], optional: true });
+    
     Table.prototype.publishProxy("itemsPerPage", "_paginator");
     Table.prototype.publishProxy("pageNumber", "_paginator", "pageNumber",1);
```
Comment (GordonSmith) [inline src/other/Table.js]: Can you move these two items to be next to the "pagination" item above as they are related.

**Comment by GordonSmith** [comment]
@jbrundage in the default theme - select a ctrl click a few rows - only column 1 gets marked as selected

**Comment by jbrundage** [comment]
@GordonSmith The default theme issue should be resolved now.

**Pull request merged**